tests/transforms/test_hamiltonian_expand.py

```diff
@@ -298,24 +298,31 @@ def test_hamiltonian_dif_tensorflow(self):
             g = gtape.gradient(res, var)
             assert np.allclose(list(g[0]) + list(g[1]), output2)
 
-    def test_processing_function_conditional_clause(self):
-        """Test the conditional logic for `len(c_group) == 1` and `len(r_group) != 1`
-        in the processing function returned by hamiltonian_expand, accessed when
-        using a shot vector and grouping if the terms don't commute with each other."""
-
-        dev_with_shot_vector = qml.device("default.qubit", shots=(10, 10, 10))
+    @pytest.mark.parametrize(
+        "H",
+        [
+            # Contains only groups with single coefficients
+            qml.Hamiltonian([1, 2.0], [qml.PauliZ(0), qml.PauliX(0)]),
+            # Contains groups with multiple coefficients
+            qml.Hamiltonian([1.0, 2.0, 3.0], [qml.X(0), qml.X(0) @ qml.X(1), qml.Z(0)]),
+        ],
+    )
+    @pytest.mark.parametrize("theta", [0.1, [0.1, 0.1, 0.1]])
+    def test_processing_function_shot_vector(self, H, theta):
```

Comment:
Can we also test that we get correct results when broadcasting, and when using broadcasting and shot vectors together? 

I can see that further up we have tests that compare the results to an expected value, but it looks to me like they are only for non-parametrized circuits. It could be a separate test here, or a new parameterization on the existing test, but I'd feel better if we test that we get good numbers out, not only that we get correctly shaped numbers.

Reply:
Done